susemanager-utils/susemanager-sls/salt-ssh/preflight.sh

```diff
@@ -202,6 +202,21 @@ elif [ "${INSTALLER}" = "apt" ]; then
     fi
 fi
 
+SELINUX_POLICY_FILENAME="salt_ssh_port_forwarding.cil"
+function selinux_policy_loaded {
+    semodule -l | grep -x $SELINUX_POLICY_FILENAME
+}
+
+# Our SSH tunnel uses a custom port and we must configure SELinux to account for it
+if [[ $REPO_HOST == "localhost" ]] && command -v selinuxenabled && selinuxenabled; then
+    if ! selinux_policy_loaded; then
+        if ! [[ -f $SELINUX_POLICY_FILENAME ]] && [[ $REPO_PORT =~ ^[0-9]+$ ]]; then
+            echo "(portcon tcp ${REPO_PORT} (system_u object_r ssh_port_t ((s0)(s0))))" >$SELINUX_POLICY_FILENAME
```

Comment:
I don't think this is the right place to check for a valid port. What happens if the port turns out to be invalid? You never write the policy module and then try to load a non-existing file?

You should place the check as soon as the port value has been set, and hard fail immediately if invalid.

Reply:
Of course, that check is in a bad place :facepalm:

Reply:
Moved the check to `REPO_PORT` initialization.